feat(complete schema discover, read stream, configure write, replicat…

MingyiLiang opened the pull request:
…ion write stream):

Commit:
feat(complete schema discover, read stream, configure write, replication write stream):

Commit:
chore(update GetpropertyTypes func and remove some unused code):

roehlerw committed:
chore: merge master

resolves #1

roehlerw commented on PluginCouchDB.Test/UnitTest1.cs:
```diff
@@ -1,13 +1,405 @@
+        }
+
+        [Fact]
+        public async Task WriteStreamTest()
```
WriteStreamReplicationTest

Commit:
chore(modified manifest and uiJson file, move test file under plogin folder):

roehlerw commented on PluginCouchDB/API/Replication/Constants.cs:
```diff
@@ -2,6 +2,6 @@ namespace PluginCouchDB.API.Replication
     public static partial class Replication
     {
-        
+        public const string Path = "localdb";
```
do you still need this file

Commit:
fix: add queryschema to manifest and update requestHelper

Commit:
feat(update manifest to support queryschema, update http request mediaType and fix some bugs):

Commit:
chore(clean up and remove unused file):

Commit:
fix(request helper, toResourceUri): update ToResourceUri method to escape special character

Commit:
feat(manifest, plugin.cs, requestHelper): add icon image to manitest, update requesthelper and plugi

Commit:
feat(support rep job from all kinds of data source):

auto generate id or user define primary key to fill _id field when write to couchdb

Commit:
fix(replication and writerecord): fix the bug that replication job stops at 226
